rust/tools/authorization-logic/raksha_examples/VisionPipelineAuthLogic

```diff
@@ -0,0 +1,153 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+"EndUser" says {
+    ownsTag("raw_video_tag").
+    ownsTag("user_selection_tag").
+    ownsTag("product_id_tag").
+      
+    // These lines say the end user trusts (i.e., delegates to) the TPPR to
+    // faithfully say when a data transformation from raw camera video to the 
+    // name of a product has taken place, but only when the user has selected 
+    // an object.
+    //
+    // It might look like this ideally:
+    //
+    // "ThirdPartyPrivacyReviewer" canSay becomes(pathX, "raw_video_tag",
+    //              "product_id_tag") :-
+    //     objectSelected("EndUser", pathX),
+    //     isAccessPath(pathX).
+    //
+    // For now it has to be done like this:
+    "ThirdPartyPrivacyReviewer" canSay declassifies(pathX, "raw_video_tag") :-
+        hasTag(pathX, "product_id_tag"),   // raw_video_tag is only removed if 
+        objectSelected("EndUser", pathX),  // product_id_tag is added
+        isAccessPath(pathX).
+
+    "ThirdPartyPrivacyReviewer" canSay
+        objectSelected("EndUser", "identified_product").
+
+    // Delegates TPPR the ability to apply tags anywhere
+    "ThirdPartyPrivacyReviewer" canSay hasTag(x, y) :- 
```

Comment:
Shouldn't this `tag` also be owned by `EndUser`?

Reply:
If that's the behavior we want, it doesn't need to be written in this policy code because this is already checked in Raksha's analysis.  
` X says hasTag(path, tag)` becomes `saysHasTag(x, path, tag)` (modulo camel->snake :) and this fact doesn't do anything unless X owns the tag (or there is some delegation like the one you are commenting on). In the Raksha implementation this is true because in 
authorization_logic.dl there is the line
`claimHasTag(principal, path, tag) :- saysHasTag(principal, path, tag).`
and in `taint.dl` there is the line `hasTag(tgt, tag) :- claimHasTag(principal, tgt, tag), ownsTag(principal, tag).`
which checks for ownership.

(I think checking this is not really needed though because the taint analysis estimates whether a handle contains sensitive data, and adding more sensitive data can only make it more likely that the code is rejected by the policy. So it is always safe to say there is more sensitive data on a handle).